### PR TITLE
Separated file_put into subtests

### PR DESCRIPTION
**Pull request by Bento007**
- split test into subtests
- added tests for invalid uuids

### Test plan
- run unittests

### Release notes
- improved test coverage for Put /file

### CODE REVIEW

File: tests/test_file.py [hunk 1]
```diff
@@ -68,22 +68,39 @@ def _test_file_put(self, replica: Replica, scheme: str, test_bucket: str, upload
...
+
+        with self.subTest(f"{replica}: Created returned when uploading a file with same payload, and different FQID"):
+            self.upload_file(source_url, str(uuid.uuid4()))
+
+        with self.subTest(f"{replica}: OK returned when uploading a file with the same payload, UUID,  version"):
+            self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid,
+                             version=version, expected_code=requests.codes.ok)
+
+        with self.subTest(f"{replica}: OK returned when uploading a file with a different payload and same FQID"):
+            self.upload_file(source_url, file_uuid, version=version, expected_code=requests.codes.ok)
```
Comment (mweiden) [inline tests/test_file.py]: How is this a different payload? (This isn't a change in your PR, but something I just now noticed)

File: tests/test_file.py [hunk 2]
```diff
@@ -68,22 +68,39 @@ def _test_file_put(self, replica: Replica, scheme: str, test_bucket: str, upload
...
+        with self.subTest(f"{replica}: Bad returned when uploading a file with an invalid version"):
+            self.upload_file(source_url, file_uuid, version='', expected_code=requests.codes.bad)
 
-        # should be able to do this twice (i.e., same payload, same UUIDs)
-        self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid,
-                         version=version, expected_code=requests.codes.ok)
+        invalid_version = 'ABCD'
+        with self.subTest(f"{replica}: server_error returned "
+                          f"when uploading a file with invalid version {invalid_version}"):
+            self.upload_file(source_url, file_uuid, version=invalid_version, expected_code=requests.codes.server_error)
```
Comment (mweiden) [inline tests/test_file.py]: The return code is changed from `bad` to `server_error`?

Comment (Bento007) [reply tests/test_file.py]: there is an issue open for this #1434

File: tests/test_file.py [hunk 3]
```diff
@@ -68,45 +68,82 @@ def _test_file_put(self, replica: Replica, scheme: str, test_bucket: str, upload
...
+        replicas = [(Replica.aws, S3Uploader, self.s3_test_bucket),
+                    (Replica.gcp, GSUploader, self.gs_test_bucket)]
+        test_data = os.urandom(max(file_sizes))
+
+        def upload_callable_creator(uploader_class: type) -> typing.Callable[[str, str, bytes], None]:
+            def upload_callable(bucket: str, key: str, data: bytes) -> None:
                 tempdir = tempfile.gettempdir()
                 uploader = uploader_class(tempdir, bucket)
-
-                src_data = os.urandom(AWS_MIN_CHUNK_SIZE + 1)
```
Comment (Bento007) [inline tests/test_file.py]: @ttung

Comment (Bento007) [reply tests/test_file.py]: before we were only testing file sizes above `AWS_MIN_CHUNK_SIZE`

**Pull request merged**